lib/utils/group-helper.js

```diff
@@ -1,6 +1,22 @@
 const { resolve, join } = require('path');
 const { existsSync } = require('fs');
 
+const parseAndValidatedFilename = pathToResolve => {
+    const pathToArray = pathToResolve.split('/');
+    let pathBeforeFileName;
+    let fileName = pathToArray[pathToArray.length - 1];
+    const isLastNameAFile = existsSync(resolve(join(pathToArray.slice(0, pathToArray.length - 1).join('/'), fileName.includes('.js') ? fileName : fileName + '.js')));
+    fileName = isLastNameAFile ? fileName : 'index.js';
+    fileName = fileName.includes('.js') ? fileName : fileName + '.js';
+    const isPathNotHavingIndexFile = pathToArray[pathToArray.length - 1] != 'index' && pathToArray[pathToArray.length - 1] != 'index.js';
```

Comment:
```js
    const isPathNotHavingIndexFile = ['index', 'index.js'].every(file => pathToArray.slice(-1).pop() !== file)
```
What do you say about this version?

Reply:
> `   const isPathNotHavingIndexFile = ['index', 'index.js'].every(file => pathToArray.slice(-1).pop() !== file)`

it looks good but I guess as far as time factor is concerned, this is more functionally heavy than the current one as there are too many calls and we can do that even just jumping to those indexes.
What do you think?

Reply:
I think the proposed solution is cleaner

Reply:
Sure ! will change it soon

Reply:
done